tests/test_decoders.py

```diff
@@ -87,3 +88,32 @@ def test_decoding_errors(header_value):
     with pytest.raises(httpcore.exceptions.DecodingError):
         response = httpcore.Response(200, headers=headers, content=compressed_body)
         response.content
+
+
+@pytest.mark.parametrize("header_value", (b"deflate", b"gzip"))
+def test_zlib_flush_errors(header_value):
+    body = b"test 123"
+    headers = [(b"Content-Encoding", header_value)]
+    mock_decompressor = mock.Mock()
+    mock_decompressor.decompress.return_value = body
+    mock_decompressor.flush.side_effect = zlib.error
+    with mock.patch(
+        "httpcore.decoders.zlib.decompressobj", return_value=mock_decompressor
+    ):
+        with pytest.raises(httpcore.exceptions.DecodingError):
+            response = httpcore.Response(200, headers=headers, content=body)
+            response.content
```

Comment:
I think we probably ought to just `pragma: nocover` the flush lines - there's no great way to test for errors in that corner, and monkeypatching to hit those lines seems like overkill.